Mathlib/MeasureTheory/Measure/DiracProba.lean

```diff
@@ -0,0 +1,196 @@
+/-
+Copyright (c) 2024 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import Mathlib.Topology.CompletelyRegular
+import Mathlib.MeasureTheory.Measure.ProbabilityMeasure
+
+/-!
+# Dirac deltas as probability measures and embedding of a space into probability measures on it
+
+## Main definitions
+* `diracProba`: The Dirac delta mass at a point as a probability measure.
+
+## Main results
+* `embedding_diracProba`: If `X` is a completely regular T0 space with its Borel sigma algebra,
+  then the mapping that takes a point `x : X` to the delta-measure `diracProba x` is an embedding
+  `X ↪ ProbabilityMeasure X`.
+
+## Tags
+probability measure, Dirac delta, embedding
+-/
+
+open Topology Metric Filter Set ENNReal NNReal BoundedContinuousFunction
+
+open scoped Topology ENNReal NNReal BoundedContinuousFunction
+
+lemma CompletelyRegularSpace.exists_BCNN {X : Type*} [TopologicalSpace X] [CompletelyRegularSpace X]
+    {K : Set X} (K_closed : IsClosed K) {x : X} (x_notin_K : x ∉ K) :
+    ∃ (f : X →ᵇ ℝ≥0), f x = 1 ∧ (∀ y ∈ K, f y = 0) := by
```

Comment:
This looks generally useful and not specific to Dirac measures. Is there an earlier file about bounded continuous functions where it would fit?

Reply:
I agree that would be desirable. But I have not found a great place. For example:
 * `Mathlib.Topology.UrysohnsBounded` does not `import Mathlib.Topology.CompletelyRegular`, so placing this there would cause a bit of import creep and this is not strictly Urysohn's lemma (although somewhat similar)
 * `Mathlib.Topology.CompletelyRegular` does not `import Mathlib.Topology.ContinuousFunction.Bounded`, so placing this there would cause a bit of import creep. This lemma is, however, quite literally a rephrasing of the property `CompletelyRegularSpace`.

Adding one of the above imports would be a possible approach, if some import creep is acceptable. Creating yet another file would be another approach, but for a single lemma this seems questionable.

I feel like I have also before scattered some results about bounded continuous functions in different files because of similar import-hierarchy problems. I have earlier even created the files:
 * `Mathlib.MeasureTheory.Integral.BoundedContinuousFunction` for integration results for bounded continuous functions.
 * `Mathlib.MeasureTheory.HasOuterApproxClosed` which contains an essentially topological property which is however meant to be used in measure theory context.

If there was a more appropriate file in the topology folder (the above ones are in the measure theory folder), a few lemmas from those files could almost certainly also be moved there. 

For the moment, however, I don't see great solutions. Perhaps a subsequent PR could address the organization of such lemmas. (Maybe also the `section NonnegativePart` that I added in `Mathlib.Totology.ContinuousFunction.Bounded` could be split off from the main bounded continuous function file.) Unfortunately the necessary imports in each of these case are slightly different, so I still don't quite know what would be the best organization.

Reply:
I'll merge master so as to get #12559 : having the subtraction of `NNReal`-valued functions makes this lemma significantly shorter (it addresses the TODO I had in the proof).